### PR TITLE
Add documentation for retry on idempotent methods

### LINKED ISSUE #14448: [HttpClient] Parameterize list of retryable methods
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#38426
| PR author(s) | @jderusse
| Merged in    | 5.x

### PR DESCRIPTION
**Pull request by jderusse**
Fix #14448

Add documentation about retry on indempotent status codes

### CODE REVIEW

File: reference/configuration/framework.rst [hunk 1]
```diff
@@ -956,13 +959,16 @@ value must use the format ``['header-name' => 'value0, value1, ...']``.
...
+**type**: ``array`` **default**: empty
 
 .. versionadded:: 5.2
 
     The ``http_codes`` option was introduced in Symfony 5.2.
 
 The list of HTTP status codes that triggers a retry of the request.
+When an empty array is provided, the default values from
+:method:`Symfony\\Component\\HttpClient\\Retry\\GenericRetryStrategy::DEFAULT_RETRY_STATUS_CODES`.
+is used.
```
Comment (wouterj) [inline reference/configuration/framework.rst]: What about this? :)

```suggestion
**type**: ``array`` **default**: :method:`Symfony\\Component\\HttpClient\\Retry\\GenericRetryStrategy::DEFAULT_RETRY_STATUS_CODES`

.. versionadded:: 5.2

    The ``http_codes`` option was introduced in Symfony 5.2.

The list of HTTP status codes that triggers a retry of the request.
```

**Review by wouterj — APPROVED** [review]

File: reference/configuration/framework.rst (hunk 1 above)
Comment (jderusse) [reply reference/configuration/framework.rst]: much better! :heart: thank you

**Comment by JohJohan** [comment]
I think the merge should go to master branch

**Comment by javiereguiluz** [comment]
Thanks Jérémy for this nice feature and for updating the docs. While merging we did some minor rewords because some long phrases were a bit difficult to read.

**Pull request merged**